src/components/SourceCodeEditor.tsx

```diff
@@ -49,33 +48,29 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
    * Returns JSON to be displayed in the visualization
    */
   const handleChanges = (incomingData: string) => {
-    if (sourceCode !== incomingData) {
-      setSourceCode(incomingData);
-
-      // update integration JSON state with changes
-      fetchIntegrationJson(incomingData, settings.dsl.name)
-        .then((res: IIntegration) => {
-          let tmpInt = res;
-          if (typeof res.metadata?.name === 'string' && res.metadata.name !== '') {
-            settings.name = res.metadata.name;
-            setSettings({ name: res.metadata.name });
-          }
-          tmpInt.metadata = { ...res.metadata, ...settings };
-          updateIntegration(tmpInt);
-        })
-        .catch((e) => {
-          console.error(e);
-        });
-    }
+    // update integration JSON state with changes
+    fetchIntegrationJson(incomingData, settings.dsl.name)
+      .then((res: IIntegration) => {
+        let tmpInt = res;
+        if (typeof res.metadata?.name === 'string' && res.metadata.name !== '') {
+          settings.name = res.metadata.name;
+          setSettings({ name: res.metadata.name });
+        }
+        tmpInt.metadata = { ...res.metadata, ...settings };
```

Comment:
Eventually we're gonna have to stop mutating so much data to prevent bugs :( this is my fault though for starting this lol